staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -167,3 +167,52 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32 `json:"samplingRatePerMillion,omitempty" protobuf:"varint,2,opt,name=samplingRatePerMillion"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityAndFairnessConfiguration holds API Priority and Fairness parameters.
+type PriorityAndFairnessConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// WorkEstimator holds work estimator parameters.
+	// +optional
+	WorkEstimator WorkEstimatorConfiguration `json:"workEstimator,omitempty"`
+}
+
+// WorkEstimatorConfiguration holds work estimator parameters.
+type WorkEstimatorConfiguration struct {
+	// ListWorkEstimator holds work estimator parameters related to list requests.
+	// +optional
+	ListWorkEstimator ListWorkEstimatorConfiguration `json:"listRequests,omitempty"`
+	// MutatingWorkEstimator holds work estimator parameters
+	// related to requests that mutate watched objects.
+	MutatingWorkEstimator MutatingWorkEstimatorConfiguration `json:"mutatingRequests,omitempty"`
+
+	// MaximumSeats is the maximum number of seats a request can occupy
+	// +optional
+	MaximumSeats *uint64 `json:"maximumSeats,omitempty"`
+}
+
+// ListWorkEstimatorConfiguration holds work estimator parameters related to list requests.
+type ListWorkEstimatorConfiguration struct {
+	// ObjectsPerSeat is a number of objects from a single list request per seat.
+	// +optional
+	ObjectsPerSeat *float64 `json:"objectsPerSeat,omitempty"`
+}
+
+// MutatingWorkEstimatorConfiguration holds work estimator
+// parameters related to watches of mutating objects.
+type MutatingWorkEstimatorConfiguration struct {
+	// EventAdditionalDuration is the additional amount of time that a mutating request occupies seats
+	// to account for the work involved in sending the corresponding WATCH notifications. Setting its
+	// value to 0 effectively disables this behavior but a request may still occupy seats.
+	// +optional
+	EventAdditionalDuration *metav1.Duration `json:"eventAdditionalDuration,omitempty"`
+	// WatchesPerSeat is the ratio between the number of WATCHes that will get notified and
+	// the number of seats occupied for EventAdditionalDuration to account for the work of
+	// sending those notifications. Setting its value to 0 disables mutating work estimator.
+	// Setting its value to a number higher than any number of watches disables this behavior but
+	// a request may still occupy a seat as per EventAdditionalDuration.
```

Comment:
Sorry, I may have made an incorrect remark earlier.  As can be seen at https://github.com/kubernetes/apiserver/blob/138386e8deaa3649caa0f2b2340069b24f004ac0/pkg/util/flowcontrol/request/mutating_work_estimator.go#L77-L129 , when WatchesPerSeat is higher than the number of watches the second phase of execution --- the extra delaying to account for WATCH notifications --- is entirely disabled; the returned value is the same as if `!e.config.Enabled` (see https://github.com/kubernetes/apiserver/blob/138386e8deaa3649caa0f2b2340069b24f004ac0/pkg/util/flowcontrol/request/mutating_work_estimator.go#L44-L47).